### ISSUE TITLE
“No Reponse” after update to 1.05 and iOS 13.2

### ISSUE THREAD

**Issue by DMBlakeley** [iRayanKhan/homebridge-chamberlain#68]
# Explain the issue occuring:
Home app on iPhone indicates “No Response” for garage door.

# Expected Result
“Open” or “Closed” expected

# Info
iOS Version: 13.2
Plugin Version: 1.06
HomeBridge Version: 0.4.50
Node/Npm Version: 5.10.0

# Log Output:
Keep getting following error in Homebridge log:

[ 2019-10-29 19:26:40] [Garage Door] SyntaxError: Unexpected token < in JSON at position 0
    at JSON.parse (<anonymous>)
    at fetch.then.then.data (/usr/local/lib/node_modules/homebridge-chamberlain/src/api.js:33:19)
    at <anonymous>
    at process._tickCallback (internal/process/next_tick.js:188:7)

Edited by iRayanKhan:
(Removed issue template gunk from Feature-Request)

**Comment by garymh** [comment]
Same here 😕

**Comment by tastheimer** [comment]
I also see this error. I have found no correlation and action I see it as soon as homebridge is restarted.
`[10/29/2019, 11:40:32 PM] [Left Garage Door] SyntaxError: Unexpected token < in JSON at position 0
    at JSON.parse (<anonymous>)
    at fetch.then.then.data (/homebridge/node_modules/homebridge-chamberlain/src/api.js:33:19)
    at process._tickCallback (internal/process/next_tick.js:68:7)
[10/29/2019, 11:40:32 PM] [Right Garage Door] SyntaxError: Unexpected token < in JSON at position 0
    at JSON.parse (<anonymous>)
    at fetch.then.then.data (/homebridge/node_modules/homebridge-chamberlain/src/api.js:33:19)
    at process._tickCallback (internal/process/next_tick.js:68:7)`

**Comment by tekie99** [comment]
same as well..

**Comment by dswee999** [comment]
Same here. Just got it working again last night from the last issue.

**Comment by nflute** [comment]
Same here after restating Homebridge

**Comment by snoslicer8** [comment]
I get the same issue starting today.

**Comment by tarbychark** [comment]
Same issue starting today as well.

**Comment by iRayanKhan** [comment]
I might have broken something in the latest update. Going to tackle it tonight. Everyone that can repo, can you paste your logs and config please? It will help pin point the issue. 

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Comment by A2-Dev** [comment]
from homebridge-error.log:
>>>
[10/29/2019, 7:26:13 PM] [MonkeyGarage] SyntaxError: Unexpected token < in JSON at position 0
    at JSON.parse (<anonymous>)
    at fetch.then.then.data (/usr/local/lib/node_modules/homebridge-chamberlain/src/api.js:33:19)
    at processTicksAndRejections (internal/process/task_queues.js:86:5)
>>>

**Comment by tarbychark** [comment]
^ Exact same error for me (different garage name of course)

**Comment by A2-Dev** [comment]
changed line 33 to
>throw new Error(data);

which returned the following: 
>[10/29/2019, 7:41:52 PM] [MonkeyGarage] Error: <html><head><title>Request Rejected</title></head><body>The requested URL was rejected. Please consult with your administrator.<br><br>Your support ID is: xXXxXxXXXxxxXXX<br><br><a href='javascript:history.back();'>[Go Back]</a></body></html>
    at fetch.then.then.data (/usr/local/lib/node_modules/homebridge-chamberlain/src/api.js:33:8)
    at processTicksAndRejections (internal/process/task_queues.js:86:5)

*edit*
ps: FreeBSD, npm and pm2 both fully updated

**Comment by markcarroll** [comment]
Same issue. Same error. 
```
[10/29/2019, 6:33:55 PM] [Garage Double Door] SyntaxError: Unexpected token < in JSON at position 0
    at JSON.parse (<anonymous>)
    at fetch.then.then.data (/homebridge/node_modules/homebridge-chamberlain/src/api.js:33:19)
    at process._tickCallback (internal/process/next_tick.js:68:7)
```
I tried removing the deviceIDs to see if it was the same issue as last time but it didn't help.

**Comment by blakems** [comment]
Here are my log outputs. I did update to 1.06 and still have the same issue.  
``` [2019-10-29 20:33:50] [Garage Door 2] SyntaxError: Unexpected token < in JSON at position 0
0|homebridge  |     at JSON.parse (<anonymous>)
0|homebridge  |     at fetch.then.then.data (/usr/lib/node_modules/homebridge-chamberlain/src/api.js:33:19)
0|homebridge  |     at <anonymous>
0|homebridge  |     at process._tickCallback (internal/process/next_tick.js:189:7)
0|homebridge  | [2019-10-29 20:33:50] [Garage Door 1] SyntaxError: Unexpected token < in JSON at position 0
```

**Comment by jocamero** [comment]
> from homebridge-error.log:
> 
> > > >
> 
> [10/29/2019, 7:26:13 PM] [MonkeyGarage] SyntaxError: Unexpected token < in JSON at position 0
> at JSON.parse ()
> at fetch.then.then.data (/usr/local/lib/node_modules/homebridge-chamberlain/src/api.js:33:19)
> at processTicksAndRejections (internal/process/task_queues.js:86:5)

Getting same error as A2-Dev

**Comment by iRayanKhan** [comment]
> changed line 33 to
> 
> > throw new Error(data);
> 
> which returned the following:
> 
> > [10/29/2019, 7:41:52 PM] [MonkeyGarage] Error: <title>Request Rejected</title>The requested URL was rejected. Please consult with your administrator.Your support ID is: xXXxXxXXXxxxXXX[Go Back]
> > at fetch.then.then.data (/usr/local/lib/node_modules/homebridge-chamberlain/src/api.js:33:8)
> > at processTicksAndRejections (internal/process/task_queues.js:86:5)
> 

Thanks @A2-Dev for this info. I wonder if it was my code change that broke it, or an API change. Last I checked my garages worked fine using the V4 API from V1.0.4. I wonder if the V5 Api went down or is being pulled temporarily. I will crack at this some more and release V1.0.7 asap.

**Comment by iRayanKhan** [comment]
Checked #67, seems like an API change. If anyone can test that branch and report back if it works I'll merge it in!

**Comment by antoniokly** [comment]
same error here

**Comment by monteroman** [comment]
Getting this with 2 doors:
SyntaxError: Unexpected token < in JSON at position 0
    at JSON.parse (<anonymous>)
    at /usr/lib/node_modules/homebridge-chamberlain/src/api.js:33:19
    at processTicksAndRejections (internal/process/task_queues.js:89:5)

This keeps repeating over and over.  

So yup, same issue.  :(

**Comment by JamesCaudill** [comment]
@iRayanKhan I tried the fix, the user agent change, and it works for me now.  I had to remove the bridge and re-add it to the Home app, and it seems to be working for me now.

**Comment by monteroman** [comment]
> Getting this with 2 doors:
> SyntaxError: Unexpected token < in JSON at position 0
> at JSON.parse ()
> at /usr/lib/node_modules/homebridge-chamberlain/src/api.js:33:19
> at processTicksAndRejections (internal/process/task_queues.js:89:5)
> 
> This keeps repeating over and over.
> 
> So yup, same issue. :(

After looking through the various threads here, I tried the user-agent change in the api.js file to say 'User-Agent': 'myQ/14041 CFNetwork/1107.1 Darwin/19.0.0' and restarted HomeBridge and it's working again for me.  I'm no longer seeing the errors.

Btw, HomeBridge is running in a docker under UnRaid.

**Comment by hemijj** [comment]
User agent change in api.js to 'User-Agent': 'myQ/14041 CFNetwork/1107.1 Darwin/19.0.0' fixed mine as well. Thanks

**Comment by tekie99** [comment]
User agent fixed mine as well! Thanks!

**Comment by DMBlakeley** [comment]
User agent change in api.js to 'User-Agent': 'myQ/14041 CFNetwork/1107.1 Darwin/19.0.0' fixed mine as well. Thanks!

**Comment by iRayanKhan** [comment]
Merged #67. I can't push it to NPM for another 7-8 hours since I don't have access to a machine with NPM to push with. Will definitely push ASAP. In the mean time you can ```git clone``` this repo or modify the ```api.js``` file.

**Comment by jwojo74** [comment]
I tried modifying api.js as suggested about after a fresh install.  I am now getting the following error.  I have two garage doors and they both get the same thing.  Any thoughts?  Thank you

Oct 30 10:19:04 raspberrypi homebridge[2893]: [2019-10-30 10:19:04] [Volkswagen Garage Door] Error: invalid response, got HTTP 500: {"code":"500.000","message":"An error occurred while processing the request.","description":"Server Error"}
Oct 30 10:19:04 raspberrypi homebridge[2893]: at res.text.then.body (/usr/local/lib/node_modules/homebridge-chamberlain/src/api.js:27:15)
Oct 30 10:19:04 raspberrypi homebridge[2893]: at <anonymous>
Oct 30 10:19:04 raspberrypi homebridge[2893]: at process._tickCallback (internal/process/next_tick.js:188:7)

**Comment by A2-Dev** [comment]
Issue seems resolved with the user agent change.
Also had the 500 issue due to deviceId but figured that out before i saw the reference.

Thanks for the work you do for this plugin,

**Comment by iRayanKhan** [comment]
@jwojo74 see #64 or the Common Issues page on the Wiki.

**Comment by jdberry** [comment]
I can confirm that changing the user agent in the api.js file fixed the issue for me.

**Comment by muzzymate** [comment]
Thank you. Changed the user agent in the api.js file to the suggested value, and now it's working again. Simply had to restart the Homebridge instance after making the change.

**Comment by gabors** [comment]
Thanks guys, updating the UA string as per above fixed this for me as well.

**Comment by johnshumate** [comment]
Changing to 'User-Agent': 'myQ/14041 CFNetwork/1107.1 Darwin/19.0.0' fixed it for me as well.  Thanks!

**Comment by ldhall1856** [comment]
@iRayanKhan when do you think you can push out that update?  

Thanks,
ldhall1856

**Comment by iRayanKhan** [comment]
Published ```V1.0.7``` with the fix to NPM. Sorry for the delay!

**Comment by iRayanKhan** [comment]
Patch is confirmed working, going to close this.
